### ISSUE TITLE
Exercise 8: failing due to fulfillment with function call?

### ISSUE THREAD

**Issue by awitherow** [stevekane/promise-it-wont-hurt#119]
I have seen that the correct solution is returning a value, then passing that value to the attachTitle function, then console logging.

I did this approach and am failing for the reason shown also below.

```javascript
require('es6-promise');

('use strict');

function attachTitle(arg) {
    return 'DR. ' + arg;
}

var manhattan = Promise.resolve(attachTitle('MANHATTAN')).then(console.log);
```
```
Your submission results compared to the expected:

                 ACTUAL                                 EXPECTED
────────────────────────────────────────────────────────────────────────────────

   "DR. MANHATTAN"                     ==    "DR. MANHATTAN"
   ""                                  ==    ""

────────────────────────────────────────────────────────────────────────────────

✓ Submission results match expected
✗ You didn’t create or didn’t use attachTitle method

# FAIL

Your solution to Values and promises didn't pass. Try again!
```

This IS returning a value at the end of the day, which is the point of the lesson?

**Comment by awitherow** [comment]
Also fails with

```javascript
require('es6-promise');

('use strict');

function attachTitle(word) {
    return 'DR. ' + word;
}

var manhattan = new Promise(function(fulfill, reject) {
    fulfill(attachTitle('MANHATTAN'));
}).then(console.log);
```

I saw the issue https://github.com/stevekane/promise-it-wont-hurt/issues/110 but I seem to be declaring a function with a proper name?

**Comment by devserkan** [comment]
The point is here I think, workshop wants to show us although our function does not return a promise, onFulfill handler wraps it in a promise. So, this is why workshop expects us to use a function which does not return a promise and chain it with .then like it turns one! You are using your function (even invoke it) in resolve, so this is very usual. You are resolving a value there (from your function). Thus, you only chain two pieces: A promise and another function.

To sum up, workshop wants to emphasize if I don't understand wrong is that:

Promise -> .then -> function not returning promise -> but again .then -> another fun

**Comment by awitherow** [comment]
Alright thanks for the clarification!
